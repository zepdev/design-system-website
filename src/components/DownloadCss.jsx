import React from 'react'
import withStyles from 'react-jss'
import Button from './button/Button'

const styles = theme => ({
  root: {
    marginBottom: theme.spacing.l.px * 2,
  },
})

const Download = ({ classes }) => {
  const handleDownload = () => {
    // const data = null
    // fetch(
    //   'https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/css/zeppelin-element-library.css',
    //   {
    //     mode: 'no-cors',
    //   }
    // )
    //   .then(resp => resp.json())
    //   .then(function(datum) {
    //     let data = datum
    //   })
    // const filename = 'zeppelin-element-library.css'
    // const blob = new Blob([data], { type: 'application/octet-stream' })
    // if (typeof window.navigator.msSaveBlob !== 'undefined') {
    //   // IE workaround for "HTML7007: One or more blob URLs were
    //   // revoked by closing the blob for which they were created.
    //   // These URLs will no longer resolve as the data backing
    //   // the URL has been freed."
    //   window.navigator.msSaveBlob(blob, filename)
    // } else {
    //   let blobURL = window.URL.createObjectURL(blob)
    //   let tempLink = document.createElement('a')
    //   tempLink.style.display = 'none'
    //   tempLink.href = blobURL
    //   tempLink.setAttribute('download', filename)

    //   // Safari thinks _blank anchor are pop ups. We only want to set _blank
    //   // target if the browser does not support the HTML5 download attribute.
    //   // This allows you to download files in desktop safari if pop up blocking
    //   // is enabled.
    //   if (typeof tempLink.download === 'undefined') {
    //     tempLink.setAttribute('target', '_blank')
    //   }

    //   document.body.appendChild(tempLink)
    //   tempLink.click()
    //   document.body.removeChild(tempLink)
    //   window.URL.revokeObjectURL(blobURL)
    // }
  }
  return (
    <div className={classes.root}>
      <Button color="primary" onClick={handleDownload}>
        Download
      </Button>
    </div>
  )
}

export default withStyles(styles)(Download)
