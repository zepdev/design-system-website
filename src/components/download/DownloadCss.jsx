import React from 'react'
import withStyles from 'react-jss'
import Button from '../button/Button'

const styles = theme => ({
  root: {
    marginBottom: theme.spacing.l.px * 2,
  },
})

const Download = ({ classes }) => {
  // const handleDownload = () => {
  //   let myHeaders = new Headers({
  //     'Content-Type': 'text/css',
  //   })
  //   fetch(
  //     'https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/css/zeppelin-element-library.css',
  //     {
  //       headers: myHeaders,
  //     }
  //   )
  //     .then(function(response) {
  //       console.log(response.text())
  //       return response.json()
  //     })
  //     .then(function(json) {
  //       console.log(json)
  //     })
  //     .catch(function(err) {
  //       console.log('Fetch problem: ' + err.message)
  //     })
  //   const filename = 'zeppelin-element-library.css'
  // const blob = new Blob([data], { type: 'application/octet-stream' })
  // if (typeof window.navigator.msSaveBlob !== 'undefined') {
  //   // IE workaround for "HTML7007: One or more blob URLs were
  //   // revoked by closing the blob for which they were created.
  //   // These URLs will no longer resolve as the data backing
  //   // the URL has been freed."
  //   window.navigator.msSaveBlob(blob, filename)
  // } else {
  //   let blobURL = window.URL.createObjectURL(blob)
  //   let tempLink = document.createElement('a')
  //   tempLink.style.display = 'none'
  //   tempLink.href = blobURL
  //   tempLink.setAttribute('download', filename)

  //   // Safari thinks _blank anchor are pop ups. We only want to set _blank
  //   // target if the browser does not support the HTML5 download attribute.
  //   // This allows you to download files in desktop safari if pop up blocking
  //   // is enabled.
  //   if (typeof tempLink.download === 'undefined') {
  //     tempLink.setAttribute('target', '_blank')
  //   }

  //   document.body.appendChild(tempLink)
  //   tempLink.click()
  //   document.body.removeChild(tempLink)
  //   window.URL.revokeObjectURL(blobURL)
  // }
  // }
  return (
    <div className={classes.root}>
      <Button
        color="primary"
        // onClick={handleDownload}
        target="_blank"
        href="https://cdn-zel.zepdev.net/zel/css/zeppelin-element-library.css"
        download="zeppelin-element-library.css"
      >
        Download
      </Button>
    </div>
  )
}

export default withStyles(styles)(Download)
