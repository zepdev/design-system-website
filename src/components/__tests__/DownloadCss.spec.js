import React from 'react'
import { render } from 'react-testing-library'
import DownloadCss from '../DownloadCss'

describe('DownloadCss', () => {
  it('renders correctly', () => {
    const component = render(<DownloadCss />)
    expect(component).toMatchSnapshot()
  })
})
