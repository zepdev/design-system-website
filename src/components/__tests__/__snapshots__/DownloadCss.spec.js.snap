// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadCss renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="zep-button zel-button"
        download=""
        href="../assets/styles/zeppelin-element-library.min.css"
      >
        Download
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="zep-button zel-button"
      download=""
      href="../assets/styles/zeppelin-element-library.min.css"
    >
      Download
    </button>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
